### ISSUE TITLE
Template function "required" no longer works

### ISSUE THREAD

**Issue by nicolaferraro** [helm/helm#7638]
I declare a field as required in the template:

```
{{ required "Value xxx is required!" .Values.xxx }}
```

When installing with version `3.1.0` without passing that parameter I get nothing installed, but no hints about the error:

```
helm install --dry-run ./helm/ --generate-name 
NAME: helm-1582154686
LAST DEPLOYED: Thu Feb 20 00:24:49 2020
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
HOOKS:
MANIFEST:

```

On `3.0.0` it was working correctly:

```
helm-3.0.0 install --dry-run ./helm/ --generate-name 
Error: execution error at (camel-k/templates/NOTES.txt:1:3): Value xxx is required!
```

Output of `helm version`:
```
version.BuildInfo{Version:"v3.1.0", GitCommit:"b29d20baf09943e134c2fa5e1e1cab3bf93315fa", GitTreeState:"clean", GoVersion:"go1.13.7"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.9", GitCommit:"2e808b7cb054ee242b68e62455323aa783991f03", GitTreeState:"clean", BuildDate:"2020-01-18T23:33:14Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:32:14Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}

```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
```
Minikube
```

**Comment by AndiDog** [comment]
Was working in 3.0.3 as well. Here's a way to reproduce and bisect this:

```sh
(
    set -e
    HELM=/tmp/helm/helm-3.1.0
    $HELM version
    cd /tmp
    rm -rf missing-bug
    $HELM create missing-bug
    rm -r missing-bug/templates/*
    kubectl create configmap a --from-literal=a=A --dry-run -o yaml >missing-bug/templates/a.yaml
    kubectl create configmap b --from-literal=b=B --dry-run -o yaml >missing-bug/templates/b.yaml
    $HELM template missing-bug missing-bug >/tmp/good
    printf '\n\nGOOD:\n'
    cat /tmp/good
    echo "-----------------------------------"

    sed 's/name: b/name: {{ required "someVar missing" .Values.someVar }}/' missing-bug/templates/b.yaml >missing-bug/templates/b.yaml.tmp
    mv -f missing-bug/templates/b.yaml.tmp missing-bug/templates/b.yaml

    $HELM template --set someVar=filledWithSomething missing-bug missing-bug >/tmp/alsogood
    printf '\n\nALSO GOOD:\n'
    cat /tmp/alsogood
    echo "-----------------------------------"

    $HELM template missing-bug missing-bug >/tmp/bad
    printf '\n\nBAD:\n'
    cat /tmp/bad
    echo "-----------------------------------"
)
```

The "BAD" case prints an empty manifest instead of failing.

**Comment by bacongobbler** [comment]
It was pointed out in another thread that #7596 might’ve fixed this. Can you check and confirm if this is an issue with a canary release?

**Comment by AndiDog** [comment]
It's fixed for me in 3.1.1 and master branch 🎉

**Comment by hickeyma** [comment]
Thanks for feedback @AndiDog.

Closing as it looks like it was fixed by #7596 and delivered in release 3.1.1. @nicolaferraro We can re-open if you find any issues.

**Comment by nicolaferraro** [comment]
Thanks!
